shell/platform/windows/flutter_windows_unittests.cc

```diff
@@ -90,5 +75,76 @@ TEST_F(WindowsTest, LaunchConflictingCustomEntrypoints) {
   ASSERT_FALSE(FlutterDesktopEngineRun(engine.get(), "conflictingEntrypoint"));
 }
 
+// Verify that native functions can be registered and resolved.
+TEST_F(WindowsTest, VerifyNativeFunction) {
+  auto& context = GetContext();
+  WindowsConfigBuilder builder(context);
+  builder.SetDartEntrypoint("verifyNativeFunction");
+
+  static fml::AutoResetWaitableEvent latch;
+  context.AddNativeFunction("Signal", [](Dart_NativeArguments args) {
```

Comment:
Just in case your didn't spot this already in `test_dart_native_resolver.h`, the `CREATE_NATIVE_ENTRY` macro overcomes the limitation of the function not being able to capture anything. Perhaps add an example here or convert the current uses to use that macro.

Reply:
In this case, notice how you had to use a static latch. This is fine for now but may get problematic when we repeat tests or run them in different orders.

Reply:
Even better; makes it a bit nicer to extract out the test expectations as well.

Reply:
Done.